docs/iris/example_code/General/polynomial_fit.py

```diff
@@ -0,0 +1,51 @@
+"""
+Fitting a polynomial
+====================
+
+This example demonstrates adding a polynomial fit to a 1D plot of data from
+an Iris cube.
+
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+import iris
+import iris.plot as iplt
+import iris.quickplot as qplt
+
+
+def main():
+    fname = iris.sample_data_path('A1B_north_america.nc')
+    cube = iris.load_cube(fname)
+
+    # Extract a single time cross-section at a latitude and longitude point.
+    location = next(cube.slices(['time']))
+
+    # Calculate a polynomial fit to the data at this cross-section.
+    x_points = location.coord('time').points
+    y_points = location.data
+    degree = 3
```

Comment:
I'm not convinced this is a good choice for this example. A second degree polynomial fits this data pretty well too...